core/trino-main/src/main/java/io/trino/simd/SimdSupportManager.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.simd;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.trino.FeaturesConfig;
+import io.trino.spi.SimdSupport;
+import io.trino.util.MachineInfo;
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.ShortVector;
+import oshi.hardware.CentralProcessor.ProcessorIdentifier;
+
+import java.util.Set;
+
+import static io.trino.util.MachineInfo.normalizeFlag;
+import static io.trino.util.MachineInfo.readCpuFlags;
+import static java.util.Locale.ENGLISH;
+
+@Singleton
+public final class SimdSupportManager
```

Comment:
I think we can get rid of this class and do the following:
1. Move the logic from `detectSimd()` into a static method on `SimdSupport` (returning a `SimdSupport` instance).
2. Switch `ServerMainModule` to bind: `binder.bind(SimdSupport.class).toInstance(SimdSupport.detectSupport())`;
3. Take both `FeaturesConfig` and `SimdSupport` as injection parameters in `BlockEncodingManager`.

Reply:
Also, maybe just add a constant to `SimdSupport` for `DETECTED` so that we have a static singleton instance that shows the result of auto-detection for debugging and avoiding accidentally detecting multiple times.